### PR TITLE
Add the common files.

### PR DESCRIPTION
**Pull request by nomis52**
This includes the API to fetch the UID from flash and the API to check / get
the boot option.

### CODE REVIEW

File: common/dfu_constants.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DFU message types
+ */
+enum {
+  DFU_DETATCH = 0,
```
Comment (peternewman) [inline common/dfu_constants.h]: Didn't we have this already, or have you moved it?

**Comment by peternewman** [comment]
Generally LGTM

File: common/dfu_constants.h (hunk 1 above)
Comment (nomis52) [inline common/dfu_constants.h]: We have Bootloader/firmware/src/dfu_constants.h but that's going to get removed in an upcoming PR.

Comment (peternewman) [inline common/dfu_constants.h]: Ah

**Pull request merged**